mycity/mycity/mycity_controller.py

```diff
@@ -0,0 +1,202 @@
+                # from another intent.
+                del mcd.session_attributes[intent_constants.ADDRESS_PROMPTED_FROM_INTENT]
+            else:
+                return create_set_address_intent_response(mcd)
+
+        # session_attributes = session.get("attributes", {})
+        if mcd.intent_name == "GetAddressIntent":
+            return get_address_from_session(mcd)
+        elif mcd.intent_name == "TrashDayIntent":
+            return request_user_address_response(mcd) \
+                if intent_constants.CURRENT_ADDRESS_KEY \
+                not in mcd.session_attributes \
+                else get_trash_day_info(mcd)
+        elif mcd.intent_name == "SnowParkingIntent":
+            return request_user_address_response(mcd) \
+                if intent_constants.CURRENT_ADDRESS_KEY \
+                not in mcd.session_attributes \
+                else get_snow_emergency_parking_intent(mcd)
+        elif mcd.intent_name == "GetAlertsIntent":
+            return get_alerts_intent(mcd)
+        elif mcd.intent_name == "AMAZON.HelpIntent":
+            return self.get_welcome_response()
+        elif mcd.intent_name == "AMAZON.StopIntent" or \
+                mcd.intent_name == "AMAZON.CancelIntent":
+            return self.handle_session_end_request()
+        elif mcd.intent_name == "UnhandledIntent":
+            return unhandled_intent(mcd)
+        else:
+            raise ValueError("Invalid intent")
+
+    def on_session_ended(self):
+        """
+        Called when the user ends the session.
+        Is not called when the skill returns should_end_session=true
+        """
+        print(
+            self.LOG_CLASS,
+            '[method: on_session_ended]',
+            'MyCityDataModel received:',
+            str(self._mcd)
+        )
+        return self._mcd
+        # add cleanup logic here
+
+    def get_welcome_response(self):
+        """
+        If we wanted to initialize the session to have some attributes we could
+        add those here.
+        """
+        print(
+            self.LOG_CLASS,
+            '[method: get_welcome_response]'
+        )
+        self._mcd.intent_name = "Welcome"
+        self._mcd.output_speech = \
+            "Welcome to the Boston Public Services skill. How can I help you? "
+
+        # If the user either does not reply to the welcome message or says
+        # something that is not understood, they will be prompted again with
+        # this text.
+        self._mcd.reprompt_text = \
+            "For example, you can tell me your address by saying, " \
+            "\"my address is\" followed by your address."
+        self._mcd.should_end_session = False
+        return self._mcd
+
+    def handle_session_end_request(self):
+        self._mcd.intent_name = "Boston Public Services - Thanks"
+        self._mcd.output_speech = \
+            "Thank you for using the Boston Public Services skill. " \
+            "See you next time!"
```

Comment:
This should probably be whispered by Alexa

Reply:
We should actually put a ticket in to update the data object to allow us to use the different tones/speeds/etc. -- I think that stuff requires a customized field in the response. Adding an issue for this.